intake_thredds/source.py

```diff
@@ -69,6 +69,12 @@ def __init__(
         self.urlpath = url
         if 'simplecache::' in url:
             self.metadata.update({'fsspec_pre_url': 'simplecache::'})
+        if isinstance(path, str):
+            path = [path]
+        if not isinstance(path, list):
+            raise ValueError(f'path must be list of str, found {type(path)}')
+        if not isinstance(path[0], str):
```

Comment:
I find this validation to be fragile. Shouldn't we be validating the entire list and not just the first element?

Reply:
implemented more robust